addons/website_event_track/static/src/js/service_worker.js

```diff
@@ -3,13 +3,18 @@ importScripts("/website_event_track/static/lib/idb-keyval/idb-keyval.js");
 const PREFIX = "odoo-event";
 const SYNCABLE_ROUTES = ["/event/track/toggle_reminder"];
 const CACHABLE_ROUTES = ["/web/webclient/version_info"];
+const MAX_CACHE_SIZE = 512 * 1024 * 1024; // 500 MB
+const MAX_CACHE_QUOTA = 0.5;
+const CDN_URL = __ODOO_CDN_URL__; // {string|undefined} the cdn_url configured for the website if activated
 
 const { Store, set, get, del } = idbKeyval;
 const pendingRequestsQueueName = `${PREFIX}-pending-requests`;
 const cacheName = `${PREFIX}-cache`;
 const syncStore = new Store(`${PREFIX}-sync-db`, `${PREFIX}-sync-store`);
 const cacheStore = new Store(`${PREFIX}-cache-db`, `${PREFIX}-cache-store`);
 const offlineRoute = `${self.registration.scope}/offline`;
+const scopeURL = new URL(self.registration.scope);
+const cdnURL = CDN_URL ? (CDN_URL.startsWith("http") ? new URL(CDN_URL) : new URL(`http:${CDN_URL}`)) : undefined;
```

Comment:
Change http to https to avoid mixed content warning/error because service workers are restricted to running across https for security reasons.

Reply:
Not sure...

1. the scheme is actually only there to have a valid argument to the `URL` constructor and not used after.
2. even if used, I wouldn't add `https` by default as upgrading the connection from HTTP to HTTPS is usual but not the opposite.

Reply:
Ok, I didn't see it's was just used for a URL constructor, so it's ok.

Note for the sharing knowledge, currently all major updated browser block HTTP request when we are inside an HTTPS scope and the console display that [error](
https://developers.google.com/web/fundamentals/security/prevent-mixed-content/imgs/simple-mixed-content-error.png) as you can try [here](https://googlesamples.github.io/web-fundamentals/fundamentals/security/prevent-mixed-content/simple-example.html).